torch/_functorch/aot_autograd.py

```diff
@@ -602,24 +602,23 @@ def convert(idx, x):
                     if req_subclass_dispatch:
                         fw_metadata = run_functionalized_fw_and_collect_metadata(
                             flat_fn,
-                            keep_input_mutations=aot_config.keep_inference_input_mutations
-                            and not needs_autograd,
-                            is_train=needs_autograd,
+                            keep_input_mutations=aot_config.keep_inference_input_mutations,
```

Comment:
cc @bdhirsh i dont know this part well enough

Reply:
seems ok